docs/DeveloperGuide.md

```diff
@@ -1,34 +1,124 @@
+
+The <code>Duke</code> class first receives user input from the <code>Ui</code> class before it creates a 
+<code>Parser</code> object and calls its <code>parseCommand</code> function to instantiate a 
+<code>SetBudgetCommand</code> object based on that user input.
+
+The <code>Duke</code> class then calls the <code>execute</code> method of the <code>SetBudgetCommand</code> object
+which makes another call to the <code>setBudget</code> function of the <code>Budget</code> object 
+with the amount specified by the user for the budget.
+
+The following sequence diagram below shows how the set budget feature works. Note the <code>Ui</code> class is
+omitted in the sequence diagram to emphasise on the other classes:
+
+![alt text](images/SetBudget.png)
+
+
+#### 2.2.2 Design considerations
+
+##### Aspect: Data structure to support the set budget feature
+
+- Alternative 1 (current choice): Object-oriented style with a separate class for <code>SetBudgetCommand</code>
+ 
+  - Pros: Easy to add the set budget feature without having to change the logic of the code much as each command object
+  is treated as a black box
+  
+  - Cons: Might significantly increase the code base with another class being added
+
+
+- Alternative 2: Implement set budget feature in the <code>Duke</code> class
+
+  - Pros: Will have lesser code to deal with as a new function is simply created in the <code>Duke</code> class
+  
+  - Cons: Code becomes less organised since for every other command that we have implemented, <code>Duke</code> class
+  simply executes those commands as black boxes, without worrying about their internal details
+
+
+## Appendix A: Product Scope
 ### Target user profile
 
-{Describe the target user profile}
+- Likes to cook at home and requires help keeping track of complex grocery shopping lists and
+staying within budget
+- Prefers to use computer applications as opposed to mobile applications
+- Can type fast
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+- Make grocery shopping a breeze by offering greater flexibility in managing
+shopping lists and also providing helpful features like budget tracking
+
 
-## User Stories
+## Appendix B: User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|organised home cook|be able to edit my budget|change my budget when I need to|
+|v1.0|organised home cook|delete items from the list|manage my list|
+|v1.0|frugal home cook|add a budget|so that I know how much I have to spend|
+|v2.0|frugal home cook|be notified when I cross my budget|remove some items from my list|
+|v2.0|practical home cook|be able to search for items on my list|find things easily in a long list|
```

Comment:
thanks for sorting it in order of v1,v2 etc.! easier for other members to follow the format 😄

Reply:
No worries haha!